src/Chapter1.hs

```diff
@@ -641,7 +660,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n = helper n
```

Comment:
That's a very nice idea to use a _helper_ recursive function! If you pass `abs n` instead of `n` to this function in here, you will handle negative numbers automatically, and you will call `abs` only once in the beginning.

Usually, such helper recursive function are called `go` in Haskell. You can read more about such pattern in our blog post, but don't worry to much for now :slightly_smiling_face: We cover recursion and this pattern in more detail in the next Chapter.

* https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go